Upgrade nav lib and refactor modal system

cvarley100 opened the pull request:
# Description

Updates RNN to v2 
Closes https://github.com/iotaledger/trinity-wallet/issues/383.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

cvarley100 committed:
Mobile: Update RNN

cvarley100 committed:
Merge branch 'develop' into feat/global-modal

cvarley100 committed:
Mobile: Update React Native Navigation to v2

cvarley100 committed:
Merge branch 'develop' into feat/global-modal

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Update routing and refactor

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Fix getSeed function

cvarley100 committed:
Mobile: Fix SeedStore bugs

cvarley100 committed:
Mobile: Fix page animation

cvarley100 committed:
Mobile: Fix wallet reset

cvarley100 committed:
Mobile: Fix account info errors

cvarley100 committed:
Mobile: Fix Android build

cvarley100 committed:
Mobile: Fix Android entry route

cvarley100 committed:
Update Android build config

Commit:
Reorder start up functions to fix splash screen hang on android

cvarley100 committed:
Merge pull request #409 from laumair/hotfix/app-load

Reorder start up functions to fix splash screen hang on android

rajivshah3 committed:
Update to Swift 4.2 and Xcode 10

rajivshah3 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Fix status bar colour

cvarley100 committed:
Mobile: Remove unused prop

cvarley100 committed:
Mobile: Fix Android back press

cvarley100 committed:
Merge branch 'feat/seed-storage-modules' into feat/global-modal

cvarley100 committed:
Mobile: Fix login authentication

cvarley100 committed:
Mobile: Fix Disable2FA backpress

cvarley100 committed:
Mobile: Increase Android topbar height

cvarley100 committed:
Mobile: Update react-native-modal

cvarley100 committed:
Mobile: Implement global modal HOC and refactor

cvarley100 committed:
Mobile: Finish modal refactor

rajivshah3 committed:
Fix failing HistoryModalContent test

rajivshah3 commented on src/shared/reducers/ui.js:
```diff
@@ -89,6 +94,17 @@ const initialState = {
+     */
     /**
      * Determines if wallet is checking state/health of the newly added custom node
      */
```
This is a duplicate of lines 105-107

cvarley100 committed:
Mobile: Fix iOS build

cvarley100 committed:
Merge pull request #426 from iotaledger/feat/global-modal-tests

Fix failing HistoryModalContent test

cvarley100 committed:
Mobile: Fix iPhone X* UI

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Fix iPhone X UI

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Resolve conflicts

cvarley100 committed:
Mobile: Refactor modals, StatefulDropdownAlert, SafeAreaView and fix various bugs

Commit:
Fix failing tests

cvarley100 committed:
Merge pull request #477 from laumair/hotfix/tests

Fix failing tests (mobile)

cvarley100 committed:
Mobile: Fix topbar crash

cvarley100 committed:
Mobile: Fix modal addresses bug

cvarley100 committed:
Mobile: Fix iPhoneX inactivity display

cvarley100 committed:
Mobile: Fix snapshot transition modal bug

cvarley100 committed:
Mobile: Fix iPhone X tabbar UI

cvarley100 committed:
Mobile: Fix up delete account UI

cvarley100 committed:
Mobile: Fix print topBar colours

cvarley100 committed:
Mobile: Fix Seed Reentry qr scanner

cvarley100 committed:
Mobile: Fix currency conversion placement

cvarley100 committed:
Mobile: Fix modal issues

cvarley100 committed:
Mobile: Disable SeedVault only on iPhone11

cvarley100 committed:
Mobile: Fix crash

cvarley100 committed:
Mobile: Fix change password crash

cvarley100 committed:
Mobile: Fix biometric modal

cvarley100 committed:
Mobile: Fix inactivity logout modal

cvarley100 committed:
Mobile: Fix Seed Reentry modal

Commit:
Purge persisted state on wallet reset

Purging stored state didn't remove persisted state from storage because imported persistor was undefined.
This fixes persistor's import and uses an alternative way of purging the stored state. Also, removes unnecessarily dispatched state actions for resetting state values.

cvarley100 committed:
Mobile: Fix HistoryModalContent props update

cvarley100 committed:
Mobile: Move DynamicStatusBar to HOC

cvarley100 committed:
Mobile: Fix SeedStore crash

cvarley100 committed:
Mobile: Fix Android StatusBar and fix crashes

cvarley100 committed:
Merge branch 'develop' into feat/global-modal

cvarley100 committed:
Mobile: Update Dropdown lib

Commit:
Merge feat/global-modal into hotfix/reset-wallet

cvarley100 committed:
Mobile: Fix Android release build

cvarley100 committed:
Mobile: Fix Android StatusBar colour

cvarley100 committed:
Mobile: Fix Android export success alert

cvarley100 committed:
Mobile: Fix SeedVault import modal crash

cvarley100 committed:
Mobile: Remove unnecessary hoists

cvarley100 committed:
Mobile: Remove unnecessary components

cvarley100 committed:
Mobile: Cleanup dropdown and statusbar

cvarley100 committed:
Shared: Rename function

cvarley100 committed:
Mobile: Fix Android StatusBar colour change

cvarley100 committed:
Mobile: Add delay to statusbar colour reset

cvarley100 committed:
Mobile: Add comments

cvarley100 committed:
Merge pull request #483 from laumair/hotfix/reset-wallet

Purge persisted state on wallet reset

cvarley100 committed:
Merge branch 'feat/global-modal' of https://github.com/iotaledger/trinity-wallet into feat/global-modal

cvarley100 committed:
Mobile: Minor adjustment to status bar

cvarley100 committed:
Mobile: Bug fixes

cvarley100 committed:
Mobile: Make splash icon transparent

cvarley100 committed:
Resolve conflicts

laumair commented on src/desktop/src/ui/Index.js:
```diff
@@ -15,7 +15,7 @@ import { getAccountNamesFromState } from 'selectors/accounts';
 import { updateTheme } from 'actions/settings';
 import { fetchNodeList } from 'actions/polling';
-import { disposeOffAlert, generateAlert } from 'actions/alerts';
+import { dismissAlert, generateAlert } from 'actions/alerts';
```
Good change.

laumair commented on src/desktop/src/ui/global/Polling.js:
```diff
@@ -158,16 +158,16 @@ class Polling extends React.PureComponent {
-    isFetchingChartData: state.polling.isFetchingChartData,
-    isFetchingMarketData: state.polling.isFetchingMarketData,
-    isFetchingAccountInfo: state.polling.isFetchingAccountInfo,
+    isPollingPrice: state.polling.isFetchingPrice,
```
👍

laumair commented on src/mobile/__tests__/ui/views/onboarding/SetAccountName.spec.js:
```diff
@@ -37,8 +35,8 @@ const getProps = (overrides) =>
     describe('propTypes', () => {
-        it('should require a navigator object as a prop', () => {
-            expect(SetAccountName.propTypes.navigator).toEqual(PropTypes.object.isRequired);
+        it('should require a componentID object as a prop', () => {
```
Typo. `componentId`

laumair commented on src/mobile/android/build.gradle:
```diff
@@ -35,6 +35,18 @@ subprojects {
                 buildToolsVersion '27.0.3'
             }
         }
+        if ((project.plugins.hasPlugin('android') || project.plugins.hasPlugin('android-library'))) {
```
Is this mistakenly added?

cvarley100 replied:
Nope

laumair commented on src/mobile/src/libs/SeedStore/Keychain.js:
```diff
@@ -12,7 +13,7 @@ class Keychain {
      */
     constructor(key, accountId) {
-        this.key = key.slice(0);
+        this.key = clone(key);
```
We should maybe use `cloneDeep` here. `clone` will not create a copy (in this case for typed arrays).

laumair commented on src/mobile/src/libs/keychain.js:
```diff
@@ -54,9 +54,12 @@ export const keychain = {
+        const nonce = await decodeBase64(secretBox.nonce);
+        return await openSecretBox(box, nonce, keyUInt8);
+    }
+    return null;
```
Shouldn't this `null` case be handled? For example https://github.com/iotaledger/trinity-wallet/blob/5f295370d677eb09eb311e11fd93d79ef7323d5f/src/mobile/src/libs/keychain.js#L93 & https://github.com/iotaledger/trinity-wallet/blob/5f295370d677eb09eb311e11fd93d79ef7323d5f/src/mobile/src/libs/keychain.js#L98

laumair commented on src/mobile/src/ui/components/DeleteAccountModal.js:
```diff
@@ -0,0 +1,81 @@
+                        {/*{t('areYouSure')}*/}
+                        Are you sure you want to delete
+                    </Text>
+                    <Text style={[styles.modalInfoText, { paddingBottom: height / 16 }, textColor]}>
```
`styles.modalInfoText` is not defined.

laumair commented on src/mobile/src/ui/components/DeleteAccountModal.js:
```diff
@@ -0,0 +1,81 @@
+        paddingVertical: height / 18,
+        width: width / 1.15,
+    },
+    questionText: {
```
This is an unused style prop.

laumair commented on src/mobile/src/ui/components/DualFooterButtons.js:
```diff
@@ -15,7 +17,7 @@ const styles = StyleSheet.create({
 });
 
 /** Onboarding buttons component */
```
I think this should also be updated.

laumair commented on src/mobile/src/ui/components/DynamicStatusBar.js:
```diff
@@ -1,49 +1,106 @@
-        if (!isModalActive && newProps.isModalActive) {
-            timer.setTimeout('timeout', () => this.resetStatusBarColor(), 50);
+        const { isModalActive, currentRoute } = this.props;
+        if (isModalActive !== newProps.isModalActive) {
```
I think it's worth mentioning/commenting here the reason why we're linking modal setup with status bar. 

Also, should this happen only for android?

laumair commented on src/mobile/src/ui/components/DynamicStatusBar.js:
```diff
@@ -1,49 +1,106 @@
+        const { isModalActive, currentRoute } = this.props;
+        if (isModalActive !== newProps.isModalActive) {
+            this.resetStatusBar(currentRoute);
+            timer.setTimeout('timeout', () => this.resetStatusBar(currentRoute), 400);
```
Maybe we should also clear timeout on unmount.

laumair commented on src/mobile/src/ui/components/ModalComponent.js:
```diff
@@ -0,0 +1,116 @@
+    class EnhancedComponent extends PureComponent {
+        static propTypes = {
+            /** Child component */
+            modalContent: PropTypes.string,
```
I think this should be a required prop. Because if its optional there's a chance `const ModalContent = MODAL_CONTENT[modalContent];` could be undefined.

laumair commented on src/mobile/src/ui/components/SeedVaultExportComponent.js:
```diff
@@ -193,14 +194,22 @@ class SeedVaultExportComponent extends Component {
+            this.props.goBack();
+            return timer.setTimeout(
+                'timeout',
+                () => this.props.generateAlert('success', t('exportSuccess'), t('exportSuccessExplanation')),
```
What is the reason for rendering alert behind a timeout here?

cvarley100 replied:
To await navigation action

laumair commented on src/mobile/src/ui/components/SingleFooterButton.js:
```diff
@@ -0,0 +1,75 @@
+    },
+});
+
+/** Onboarding buttons component */
```
Typo.

laumair commented on src/mobile/src/ui/routes/navigation.js:
```diff
@@ -26,52 +30,66 @@ import ForceChangePassword from 'ui/views/wallet/ForceChangePassword';
+function applyHOCs(screen) {
     if (isIPhoneX) {
-        return withSafeAreaView(screen);
+        screen = withSafeAreaView(screen);
```
Can't think the affect reassignment can cause here, but I would suggest to avoid it.

laumair commented on src/mobile/src/ui/views/onboarding/SeedReentry.js:
```diff
@@ -184,50 +181,37 @@ class SeedReentry extends Component {
-                );
-                break;
+                return this.props.toggleModalActivity(modalContent, {
+                    validatePassword: (password) => this.SeedVaultImport.validatePassword(password),
```
`SeedVaultImport` -> `seedVaultImport`?

laumair commented on src/mobile/src/ui/views/wallet/Home.js:
```diff
@@ -305,44 +290,35 @@ class Home extends Component {
             accountName: this.props.selectedAccountName,
             task: 'hasDisplayedTransitionGuide',
         });
+        if (this.props.isModalActive) {
```
I think this should rather be done directly in `componentDidMount`. It doesn't really relate to `completeTransitionTask`.

laumair commented on src/mobile/src/ui/views/wallet/Loading.js:
```diff
@@ -163,19 +169,35 @@ class Loading extends Component {
+            this.redirectToLogin();
+        }
+        if (!hasErrorFetchingFullAccountInfo && newProps.hasErrorFetchingFullAccountInfo) {
+            if (accountNames.length <= 1) {
```
What is the reason we'll directly navigate to dashboard (if there are multiple accounts)?

cvarley100 replied:
For when you're adding an additional account

laumair commented on src/shared/actions/accounts.js:
```diff
@@ -491,15 +479,13 @@ export const manuallySyncAccount = (seedStore, accountName) => {
  * @returns {function} dispatch
  */
-export const getAccountInfo = (seedStore, accountName, navigator = null, notificationFn) => {
+export const getAccountInfo = (seedStore, accountName, notificationFn) => {
```
This could affect desktop where its using this state action.

cvarley100 committed:
Make requested changes

cvarley100 committed:
Mobile: Minor fix

Commit:
Fix entry point for app

cvarley100 committed:
Merge pull request #493 from laumair/hotfix/app-entry

Fix entry point for app

laumair commented:
@rajivshah3 Can we want to update entry point to `index.js`? I believe CI build is failing because of that.

rajivshah3 commented:
@laumair the `index.js` path on Bitrise has been updated, so that should pass. I can't update the path on Buildkite yet because it requires https://github.com/iotaledger/ci-pipelines/pull/10 to be merged

rajivshah3 committed:
Update lockfile

rajivshah3 committed:
Use withNamespaces as translate is now deprecated

rajivshah3 committed:
Merge branch 'feat/global-modal' into hotfix/splashscreen

rajivshah3 committed:
Merge pull request #492 from iotaledger/hotfix/splashscreen

Hotfix/splashscreen

cvarley100 committed:
Merge branch 'develop' into feat/global-modal

cvarley100 committed:
Mobile: Fix iPhone X light theme visual bug

cvarley100 committed:
Mobile: Fix iPhone X modal positioning

cvarley100 committed:
Mobile: Add change node timeout when adding additional seed

rajivshah3 commented on src/desktop/yarn.lock:
```diff
@@ -5,24 +5,20 @@
 "7zip-bin@~4.0.2":
   version "4.0.2"
   resolved "https://registry.yarnpkg.com/7zip-bin/-/7zip-bin-4.0.2.tgz#6abbdc22f33cab742053777a26db2e25ca527179"
-  integrity sha512-XtGk+IF57pr852UK1AhQJXqmm1WmSgS5uISL+LPs0z/iAxXouMvdlLJrHPeukP6gd7yR2rDTMSMkHNODgwIq7A==
```
I think your version of `yarn` keeps removing the integrity field

cvarley100 committed:
Shared: Theme adjustments
